llvm/include/llvm/Support/YAMLTraits.h

```diff
@@ -901,11 +901,12 @@ class IO {
   }
 
   template <typename T, typename Context>
-  std::enable_if_t<has_SequenceTraits<T>::value, void>
-  mapOptionalWithContext(const char *Key, T &Val, Context &Ctx) {
-    // omit key/value instead of outputting empty sequence
-    if (this->canElideEmptySequence() && !(Val.begin() != Val.end()))
-      return;
+  void mapOptionalWithContext(const char *Key, T &Val, Context &Ctx) {
+    if constexpr (has_SequenceTraits<T>::value) {
+      // omit key/value instead of outputting empty sequence
+      if (this->canElideEmptySequence() && !(Val.begin() != Val.end()))
```

Comment:
```suggestion
      if (this->canElideEmptySequence() && Val.begin() == Val.end())
```
While here? Unless there's some weird reason it has to be written that way...

Reply:
I wonder if `!=` is to accommodate bare minimum iterators that define `operator!=`, `operator*`, and `operator++`.  Anyway, builds and tests seem to pass with `==` here.